src/SoapCore/SoapEndpointMiddleware.cs

```diff
@@ -57,7 +55,7 @@ public SoapEndpointMiddleware(ILogger<SoapEndpointMiddleware> logger, RequestDel
 
 		public async Task Invoke(HttpContext httpContext, IServiceProvider serviceProvider)
 		{
-			httpContext.Request.EnableRewind();
+			httpContext.Request.EnableBuffering();

```

Comment:
It looks like you have some bonus changes. I presume that's because you merged master into your branch instead of develop.

Reply:
I has merged master